dataprofiler/profilers/profile_builder.py

```diff
@@ -1355,11 +1359,125 @@ def _update_row_statistics(self, data, sample_ids=None):
+
+        :param clean_samples: the input cleaned dataset
+        :type clean_samples: dict()
+        """
+        if self.correlation_matrix is not None:
+            # currently return None with the exising correlation
+            # TODO: implement the update with the exising correlation
+            warnings.warn("Currently, the updated correlation is kept the same "
+                          "as the existing value. Updating correlations with "
+                          "existing value will be available in a future update")
+            return
+
+        self._get_correlation(clean_samples)
+
+    def _merge_correlation(self, other):
+        """
+        Merge correlation matrix from two profiles
+
+        :param other: the other profile that needs to be merged
+        :return:
+        """
+        corr_mat1 = self.correlation_matrix
+        mean1 = np.array(
+            [self._profile[profile_name].profile['statistics']['mean']
+             for profile_name in self._profile])
+        std1 = np.array(
+            [self._profile[profile_name].profile['statistics']['stddev']
+             for profile_name in self._profile])
+        n1 = self.total_samples
+
+        corr_mat2 = other.correlation_matrix
+        mean2 = np.array(
+            [other._profile[profile_name].profile['statistics']['mean']
+             for profile_name in other._profile])
+        std2 = np.array(
+            [other._profile[profile_name].profile['statistics']['stddev']
+             for profile_name in other._profile])
+        n2 = other.total_samples
+        return self._merge_correlation_helper(corr_mat1, mean1, std1, n1,
+                                              corr_mat2, mean2, std2, n2)
+
+    @staticmethod
+    def _merge_correlation_helper(corr_mat1, mean1, std1, n1,
+                                  corr_mat2, mean2, std2, n2):
+        """
+        Helper function to merge correlation matrix from two profiles
+
+        :param corr_mat1: correlation matrix of profile1
+        :type corr_mat1: pd.DataFrame
+        :param mean1: mean of columns of profile1
+        :type mean1: np.array
+        :param std1: standard deviation of columns of profile1
+        :type std1: np.array
+        :param corr_mat2: correlation matrix of profile2
+        :type corr_mat2: pd.DataFrame
+        :param mean2: mean of columns of profile2
+        :type mean2: np.array
+        :param std2: standard deviation of columns of profile2
+        :type std2: np.array
+        :return: merged correlation matrix
+        """
+        if len(corr_mat1.columns) != len(corr_mat2.columns) or \
+                len(corr_mat1.columns) <= 1:
+            return None
+        if set(corr_mat1.columns).difference(set(corr_mat2.columns)):
+            return None
+
+        std_mat1 = np.outer(std1, std1)
+        std_mat2 = np.outer(std2, std2)
+        mean_diff_vector = mean1 - mean2
+        mean_diff_mat = np.outer(mean_diff_vector, mean_diff_vector)
```

Comment:
what if the column order changed? Wouldn't this be an issue? Not saying it did, but what if? Do we want to map or have an auto block on even schema order change?

Reply:
good point, i've added a fix to that, please look at the merge_correlation function